src/grpc/stream_infer_handler.cc

```diff
@@ -745,7 +747,9 @@ ModelStreamInferHandler::StreamInferResponseComplete(
       }
     } else {
       state->step_ = Steps::WRITEREADY;
-      state->context_->WriteResponseIfReady(state);
+      if (is_complete) {
+        state->context_->WriteResponseIfReady(state);
```

Comment:
Can we add a test for a bad user-error case where the model sends two non-null response objects for a non-decoupled model? I'm not sure this is handled here. Would be good to gracefully handle it and raise an error rather than crash or overwrite state.

Reply:
Sure. Added the test case for a non-decoupled model returning more than one response during gRPC streaming. The client should get the first response.